app/scripts/modules/amazon/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -34,6 +34,9 @@ module.exports = angular.module('spinnaker.aws.serverGroupCommandBuilder.service
           var keyPair = credentials ? credentials.defaultKeyPair : null;
           var applicationAwsSettings = _.get(application, 'attributes.providerSettings.aws', {});
 
+          var defaultIamRole = settings.providers.aws.defaults.iamRole || '{{application}}InstanceProfile';
+          defaultIamRole = defaultIamRole.replace('{{application}}', application.name);
```

Comment:
you can simplify these two lines to just `const defaultIamRole = settings.providers.aws.defaults.iamRole || `${application.name}InstanceProfile`;